inc/plugin.class.php

```diff
@@ -94,21 +100,28 @@ function getFromDBbyDir($dir) {
     * @return void
    **/
    function init() {
-      global $DB, $GLPI_CACHE;
-
-      $GLPI_CACHE->set('plugins', []);
+      global $DB;
 
       if (!isset($DB) || !$DB->connected) {
          // Cannot init plugins list if DB is not connected
+         self::$loaded_plugins = [];
          return;
       }
 
-      $this->checkStates();
+      // Fill active plugins with database state then check state of each plugin to
+      // remove thoose that cannot stay active.
       $plugins = $this->find(['state' => self::ACTIVATED]);
+      self::$loaded_plugins = array_column($plugins, 'directory', 'id');
```

Comment:
I changed the logic here, to be sure that if, for any reason, something retrieve plugin list during initialization process, this list will be populated with last valid state (that was validated on previous request).